src/server/game/Battlefield/Zones/BattlefieldWG.cpp

```diff
@@ -1449,29 +1449,24 @@ void BattlefieldWG::UpdateTenacity()
     if (newStack == int32(m_tenacityStack))
         return;
 
-    if (m_tenacityStack > 0 && newStack <= 0)               // old buff was on alliance
-        team = TEAM_ALLIANCE;
-    else if (newStack >= 0)                                 // old buff was on horde
-        team = TEAM_HORDE;
-
     m_tenacityStack = newStack;
     // Remove old buff
-    if (team != TEAM_NEUTRAL)
+    if (m_tenacityTeam != TEAM_NEUTRAL)
```

Comment:
much better now, please always ensure all fields are always initialized. even if it builds and if it might look like working, uninitialized values can cause the most random bugs